### PR TITLE
Added examples, waypoint manager

### PR DESCRIPTION
**Pull request by hildebrandt-carl**
Added 2 examples as well as a waypoint manager.

### CODE REVIEW

File: freyja_controller.launch [hunk 1]
```diff
@@ -39,43 +38,42 @@
...
+      <arg name="fcu_url" value="$(arg comm_port)"/>
+    </include>
   </group>
+
+  <node name="waypoint_manager" pkg="waypoint_manager" type="waypoint_manager_node" output="screen"
+				if="$(eval use_waypoint_handler)"> 
+	</node>
+
+  <include file="$(find freyja_examples)/launch/example.launch" if="$(eval use_examples)" >
+    <arg name="example_number" value="$(arg example_number)"/>
```
Comment (ajshank) [inline freyja_controller.launch:76]: Don't see `example_number` as an argument ..

File: freyja_controller.launch [hunk 2]
```diff
@@ -29,8 +29,7 @@
 
   <!-- autostart rosbag -->
 	<arg name="start_rosbag" default="false"/>
-	
-	
+
```
Comment (ajshank) [inline freyja_controller.launch:32]: Unused arguments: `waypoint_handler_mode`, `example_trajectory`. Argument `example_number` should be announced here as well ..

File: freyja_examples/CMakeLists.txt [hunk 3]
```diff
@@ -31,18 +31,9 @@ include_directories(
...
-
-
-# vision-based guidance for approach
-# add_executable(trajectory_shaper_node src/trajectory_shaper.cpp)
-# add_dependencies(trajectory_shaper_node ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
-
-# target_link_libraries(trajectory_shaper_node
-#   ${catkin_LIBRARIES}
-# )
+ )
```
Comment (ajshank) [inline freyja_examples/CMakeLists.txt]: .. weird story here: files should end with a newline character :)
This used to be necessary for older serial terminals and file pipes, still a common practice.

File: freyja_examples/src/ref_to_point.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+
```
Comment (ajshank) [inline freyja_examples/src/ref_to_point.py:2]: Consider adding a block-comment here to briefly say what the file is for: converts to `geometry_msgs` type for visualisation purposes (rqt_) only.

File: waypoint_manager/package.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>waypoint_manager</name>
+  <version>0.0.0</version>
+  <description>Waypoint manager package</description>
+
+  <maintainer email="ajshank@noreply.users.github.com">ajshank</maintainer>
+  <author email="ajshank@noreply.users.github.com">ajshank</author>
+
```
Comment (ajshank) [inline waypoint_manager/package.xml:9]: Add another <author> tag here.

**Review by ajshank — CHANGES_REQUESTED** [review]
Added a few comments, the main launch file has some args misplaced - likely due to the final merge you did. You can add these into commits here, not a problem.

File: freyja_controller.launch (hunk 2 above)
Comment (hildebrandt-carl) [reply freyja_controller.launch:29]: Sorry, this must have been removed in the merge. Fixed

File: freyja_controller.launch (hunk 1 above)
Comment (hildebrandt-carl) [reply freyja_controller.launch:73]: Sorry, this must have been removed in the merge. Fixed

File: freyja_controller.launch (hunk 2 above)
Comment (hildebrandt-carl) [reply freyja_controller.launch:29]: I think during the merge `waypoint_handler_mode`, `example_trajectory` were added and `example_number` removed. Sorry about that.

File: freyja_examples/src/ref_to_point.py (hunk 4 above)
Comment (hildebrandt-carl) [reply freyja_examples/src/ref_to_point.py:2]: This was accidentally uploaded in this merge. I wasn't planning on adding it, I have removed it from this merge. However if you think it would be useful to include I will bring it back.

File: waypoint_manager/package.xml (hunk 5 above)
Comment (hildebrandt-carl) [reply waypoint_manager/package.xml]: I assume you mean as `maintainer` and `author`, done.

File: freyja_examples/CMakeLists.txt (hunk 3 above)
Comment (hildebrandt-carl) [reply freyja_examples/CMakeLists.txt]: Done.

Comment (hildebrandt-carl) [reply freyja_examples/CMakeLists.txt]: Also didn't know that so I learnt something today. Thanks for pointing that out :)

**Comment by hildebrandt-carl** [comment]
I have made the requested changes.

**Review by ajshank — APPROVED** [review]
Approving for merge into Freyja:waypoint-handler-devel.

**Comment by ajshank** [comment]
@hildebrandt-carl There's a bug in `waypoint_manager`'s waypoint callback at [L203](https://github.com/unl-nimbus-lab/Freyja/blob/0f867ff4cf456edf62df387435f8560b7ce8332c/waypoint_manager/src/waypoint_manager.cpp#L203). The trajectory will be initialised (`traj_init_ = true`) even if the specified mode is incorrect. You should `return` in the `else` right before it (after printing the WARN).

Even better, I strongly recommend changing the style in this callback to a "guard block" style at the top:
```
if ( bad_arguments )
{
  print warning;
  return;
}
// normal execution here.
```
Basically weed out a problematic argument early on, and block the execution there. No need for an `else` too (because of `return`), and the rest of the code can have one indent less, thus preventing what's often called "horizontal programming" .. :)

**Comment by hildebrandt-carl** [comment]
This should solve Issue #4

**Comment by hildebrandt-carl** [comment]
> @hildebrandt-carl There's a bug in `waypoint_manager`'s waypoint callback at [L203](https://github.com/unl-nimbus-lab/Freyja/blob/0f867ff4cf456edf62df387435f8560b7ce8332c/waypoint_manager/src/waypoint_manager.cpp#L203). The trajectory will be initialised (`traj_init_ = true`) even if the specified mode is incorrect. You should `return` in the `else` right before it (after printing the WARN).
> 
> Even better, I strongly recommend changing the style in this callback to a "guard block" style at the top:
> 
> ```
> if ( bad_arguments )
> {
>   print warning;
>   return;
> }
> // normal execution here.
> ```
> 
> Basically weed out a problematic argument early on, and block the execution there. No need for an `else` too (because of `return`), and the rest of the code can have one indent less, thus preventing what's often called "horizontal programming" .. :)

Done,

I opted not to go with "guard block" because the code is already inside a check to see if any bad arguments are specified. I however am not against changing that. I just wanted to avoid massive changes to code structure which I could not verify is bug free in this PR.
